From d7ba4308fff23d1003c0f842c92302df754a9783 Mon Sep 17 00:00:00 2001
From: Xianzhong <b07117@freescale.com>
Date: Tue, 5 Feb 2013 20:10:18 +0800
Subject: [PATCH 08/37] ENGR00243299 Fixed SystemUI crash when disable GPU
 acceleration

The designated logic should switch to SW Canvas render if OpenGL-ES 2.0
is not available when GPU 3D core is disabled or in-existed.

The fix is for debug purpose on i.mx6d/q devices when disable GPU 3D.
Also enable JB4.2 on i.mx6sl device without GPU 3D core.

Signed-off-by: Xianzhong <b07117@freescale.com>
---
 .../SystemUI/src/com/android/systemui/ImageWallpaper.java     | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/ImageWallpaper.java b/packages/SystemUI/src/com/android/systemui/ImageWallpaper.java
index 4b0c2cb..fa6fcd0 100644
--- a/packages/SystemUI/src/com/android/systemui/ImageWallpaper.java
+++ b/packages/SystemUI/src/com/android/systemui/ImageWallpaper.java
@@ -14,6 +14,8 @@
  * limitations under the License.
  */
 
+/* Copyright (C) 2013 Freescale Semiconductor, Inc. */
+
 package com.android.systemui;
 
 import static android.opengl.GLES20.*;
@@ -624,13 +626,16 @@ public class ImageWallpaper extends WallpaperService {
 
             mEglConfig = chooseEglConfig();
             if (mEglConfig == null) {
-                throw new RuntimeException("eglConfig not initialized");
+                //chooseEGLConfig fail since no egl render type EGL_OPENGL_ES2_BIT when disable gpu acceleration,
+                //should switch to SW canvas renderer without throwing exception in this case.
+                checkEglError();
+                return false;
             }
 
             mEglContext = createContext(mEgl, mEglDisplay, mEglConfig);
             if (mEglContext == EGL_NO_CONTEXT) {
-                throw new RuntimeException("createContext failed " +
-                        GLUtils.getEGLErrorString(mEgl.eglGetError()));
+                checkEglError();
+                return false;
             }
 
             int attribs[] = {
-- 
1.8.0

